packages/donor/src/screens/UpcommingDonation/UpcomingDonationScreen.tsx

```diff
@@ -15,19 +15,75 @@ import Cancellation from "../../assets/images/cancelation.svg";
 import Whatsapp from "../../assets/images/whatsup-color-big.svg";
 import TrashIcon from "../../assets/icons/trash.svg";
 import UpcomingDonationInfo from "./UpcomingDonationInfo";
+import makeUrls, { TCalendarEvent } from "add-event-to-calendar";
+import ICalendarLink from "react-icalendar-link";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
 
 export interface UpcomingDonationScreenProps {
   bookedAppointment: BookedAppointment;
   fullName: string;
   onCancel: () => Promise<void>;
 }
 
+interface CalendarType {
+  type: string;
+  label: string;
+}
+
+const calendarTypeList: CalendarType[] = [
+  { type: "google", label: "Google" },
+  { type: "ics", label: "Apple" },
+];
+
 export default function UpcomingDonationScreen({
   fullName,
   onCancel,
   bookedAppointment,
 }: UpcomingDonationScreenProps) {
-  const donationDate = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateStart = new Date(bookedAppointment.donationStartTimeMillis);
+  const eventDateEnd = new Date(eventDateStart.getTime() + 90 * 60000);
+  const eventLocation = `בית החולים ${LocaleUtils.getHospitalName(
+    bookedAppointment.hospital
+  )}`;
+  const eventDescription = `זכרון מנחם - ${eventLocation} - ${eventDateStart.toLocaleDateString(
+    "he-He",
+    DateUtils.ShortDateFormat
+  )}`;
+
+  const getGoogleCalendarEvent = (): TCalendarEvent => ({
+    name: `תרומת טרומבוציטים`,
+    location: eventLocation,
+    details: eventDescription,
+    startsAt: eventDateStart.toString(),
+    endsAt: eventDateEnd.toString(),
+  });
+
+  const getAppleCalendarEvent = (): any => ({
+    title: "תרומת טרומבוציטים",
+    location: eventLocation,
+    description: eventDescription,
+    startTime: eventDateStart.toString(),
+    endTime: eventDateEnd.toString(),
+  });
+
+  const [anchorEl, setAnchorEl] = React.useState<Boolean>(false);
+
+  const handleClick = () => {
+    setAnchorEl(true);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(false);
+  };
+
+  const addToCalendarByType = (calendar: CalendarType) => {
```

Comment:
Please create a dedicated Util in `common` that accepts a file type and returns a link to an invite.

This should be in common so we could use it in the backend as well (when sending emails).

Also, please add tests to it